### PR TITLE
`construct_runtime!` macro

### PR DESCRIPTION
**Pull request by KitHat**
Added a construct runtime macro implementation.
This PR is draft until the #1 is finished.

Task List:

- [x] Code part finished
- [x] Readme added
- [x] EVM abstraction added

### CODE REVIEW

**Review by ozgunozerk — COMMENTED** [review]
I'll have multiple comments, will post them in different threads so that they will be more organized. Let me start with the good:

I like that counter approach is used for pallet indices, and they are hidden from the users. 
If users were to specify pallet indices along with names, it wouldn't be much different than the current way how construct runtime macro from parity works.

However, this will be a breaking change for the projects that want to migrate to this approach. I think there isn't any solution that mitigates both problems, and I'm favoring the approach in this PR on this matter.

**Comment by ozgunozerk** [comment]
I like the approach here, the logic is similar to what I've written, glad to see good minds think alike 🚀 

Overall, I think it's good! However, if we are trying to reduce the boilerplate by introducing an opinionated configurations, I think we should provide the following:
- there should be a significant difference in terms of simplicity between the old boilerplate way, and our way. And I can envisage it can easily be much better in terms of simplicity and compactness
- having an opinionated template/setup is good, because it has probably around 60-80% of the common setups across the ecosystem. However, if we do not offer configuration to great extend, I highly suspect the adoption will be near to 0. All the projects out there have their **_own specific_** setup, and imo they should be able to change ANYTHING in our opinionated setup if they want to.

**Comment by KitHat** [comment]
> Also this doesn't depend on #1 and #1 does not depend on it so this PR could be made against the main branch, but would still prefer confirmation that it is tested against the templates prior to merging.

I have tested it against the the Generic template. Yet have to test it against the EVM template

**Comment by ozgunozerk** [comment]
I wrote the above comments 2 weeks ago but forgot to click on `review` button to submit them, 🤦 

We can address them later in another PR if we are short on time

File: procedural/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[package]
+name = "openzeppelin-procedural"
```
Comment (ggonzalez94) [inline procedural/Cargo.toml:2]: If this is too large of a change at this point feel free to disregard it, but something like `openzeppelin-polkadot-wrappers-proc` would be more consistent I think(given the main package name)

Comment (KitHat) [reply procedural/Cargo.toml:2]: It is not large at all :)
Will fix first thing tomorrow

**Comment by KitHat** [comment]
This PR is closed in favor of #21

**Pull request closed without merging**